pkg/config/network.go

```diff
@@ -1,71 +1,168 @@
 package config
 
-type AppConfig struct {
-	Network    string
-	NodeURL    string
-	DNSAddress string
-}
+import (
+	"embed"
+	"fmt"
+	"log"
+	"sync"
+
+	"gopkg.in/yaml.v2"
+)
 
 const (
-	testnetNodeURL = "https://test.massa.net/api/v2"
-	// testnet20.2.
-	testnetDNSAddress = "AS12YMz7NjyP3aeEWcSsiC58Hba8UxHapfGv7i4PmNMS2eKfmaqqC"
+	MassaStationURL = "station.massa"
+)
 
-	labnetNodeURL    = "https://labnet.massa.net/api/v2"
-	labnetDNSAddress = "AS1PV17jWkbUs7mfXsn8Xfs9AK6tHiJoxuGu7RySFMV8GYdMeUSh"
+// NetworkManager represents a manager for network configurations.
+type NetworkManager struct {
+	appConfig     AppConfig                // Current network configuration
+	knownNetworks map[string]NetworkConfig // Known network configurations
+	networkMutex  sync.RWMutex             // Mutex for concurrent access to network configuration
+}
 
-	buildnetNodeURL    = "https://buildernet.massa.net/api/v2"
-	buildnetDNSAddress = "AS12aGWkBorEM2EpKeNyigSkoCqwdxm872g5KkzUiox3v4VosFW3F"
+// Verify at compilation time that NetworkManager implements NetworkManagerInterface.
+//
+//nolint:exhaustruct
+var _ NetworkManagerInterface = &NetworkManager{}
```

Comment:
why do you need an interface ? :/ from my experience, we use it when we need to mock the object. but here i don't see the purpose